packages/dma-contracts/contracts/actions/common/CollectFee.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.15;
+
+import { Executable } from "../common/Executable.sol";
+import { UseStorageSlot, StorageSlot, Write, Read } from "../../libs/UseStorageSlot.sol";
+import { UseRegistry } from "../../libs/UseRegistry.sol";
+import { ServiceRegistry } from "../../core/ServiceRegistry.sol";
+import { SafeERC20, IERC20 } from "../../libs/SafeERC20.sol";
+
+/**
+ * @title CollectFee | Fee Collection Action Contract
+ * @notice Collects a fee based on a transaction amount read from an OperationStorage slot
+ */
+contract CollectFee is Executable, UseStorageSlot, UseRegistry {
+    using SafeERC20 for IERC20;
+    using Write for StorageSlot.TransactionStorage;
+    using Read for StorageSlot.TransactionStorage;
+
+    // Fee percentage (e.g., 1% = 100, 0.5% = 50)
+    uint256 public feePercentage;
```

Comment:
will taht work ? ar action called or delegatecalled ? I'd make it constant or immutable